cmd/convox/env.go

```diff
@@ -121,12 +129,49 @@ func cmdEnvGet(c *cli.Context) error {
 	return nil
 }
 
+// Validate environment variable keypair format; trim spaces and surrounding single quotes.
+func cleanEnvPair(value string) (string, error) {
+	// Don't let CLI swallow panics. (ref. urfave/cli/issues/451)
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Println("ERROR: ", r, string(debug.Stack()))
+		}
+	}()
+
+	parts := strings.SplitN(value, "=", 2)
+	if len(parts) != 2 {
+		return "", errors.New("Environment variables should be defined in key=value format. You specified: " + value)
+	}
+
+	if key := strings.TrimSpace(parts[0]); key != "" {
+		val := parts[1]
+		if len(val) == 0 {
+			return "", errors.New("Can't set " + key + " to an empty value; try `convox env unset`.")
```

Comment:
@ddollar I took a look at `SetEnvironment` in `client/environment.go`:

```
func (c *Client) SetEnvironment(app string, body io.Reader) (Environment, string, error) {
    var env Environment 

    res, err := c.PostBodyResponse(fmt.Sprintf("/apps/%s/environment", app), body, &env)

    if err != nil {
        return nil, "", err
    }

    return env, res.Header.Get("Release-Id"), nil
}
```

Are you suggesting we move `cleanEnvPair()` to `client/environment.go`, then modify `SetEnvironment()` to read `body` to a string, validate/manipulate it, change it back to `body` and then pass it on to `c.PostBodyResponse()`?

It feels weird to manipulate `body`, so would like to make sure I haven't misunderstood what you're saying.